### PR TITLE
Fix: Correctly determine the bus to attatch custom powerplants to.

### PR DESCRIPTION
**Pull request by euronion**
## Changes proposed in this Pull Request

This PR fixes an annoying "feature" when adding custom powerplants:
Choosing an in-use `id` for a custom powerplant added this custom powerplant to the same bus, to which existing powerplant with this index gets attatched to.

E.g. if the custom powerplant was located in Germany (according to `lat`, `lon` and `country` entries) but had `id=0` in `data/custom_powerplants.csv`, then it would end up in one of the buses in Norway (bus `6753`) where the retrieved powerplant with `id=0` is correctly located (index collission).

With this PR:

* a unique 'id' is enforced for the generated powerplant list `resources/powerplants.csv`
* the 'id' column is removed from the `data/custom_powerplants.csv` template and no longer required
* an 'id' is automatically assigned to entries from `data/custom_powerplants.csv`

The PR should be backwards compatible with `data/custom_powerplants.csv` files which were previously working (files which were not working correctly will now also throw exceptions).

## Checklist

- [x] I tested my contribution locally and it seems to work fine.
- [x] Code and workflow changes are sufficiently documented.
- [x] Newly introduced dependencies are added to `environment.yaml` and `environment.docs.yaml`.
- [x] Changes in configuration options are added in all of `config.default.yaml`, `config.tutorial.yaml`, and `test/config.test1.yaml`.
- [x] Changes in configuration options are also documented in `doc/configtables/*.csv` and line references are adjusted in `doc/configuration.rst` and `doc/tutorial.rst`.
- [ ] A note for the release notes `doc/release_notes.rst` is amended in the format of previous release notes.

### CODE REVIEW

File: data/custom_powerplants.csv [hunk 1]
```diff
@@ -1 +1 @@
-id,Name,Fueltype,Technology,Set,Country,Capacity,Efficiency,Duration,Volume_Mm3,DamHeight_m,YearCommissioned,Retrofit,lat,lon,projectID
\ No newline at end of file
+Name,Fueltype,Technology,Set,Country,Capacity,Efficiency,Duration,Volume_Mm3,DamHeight_m,YearCommissioned,Retrofit,lat,lon,projectID
```
Comment (fneum) [inline data/custom_powerplants.csv]:
```suggestion
Name,Fueltype,Technology,Set,Country,Capacity,Efficiency,Duration,Volume_Mm3,DamHeight_m,YearCommissioned,Retrofit,lat,lon,projectID,YearDecommissioning
```
Just to have all columns of powerplantmatching. It must have slipped through before.

**Comment by fneum** [comment]
Just a small note in the release notes missing. Maybe add `[skip travis]` to commit message. I forgot that in the last one.

**Comment by fneum** [comment]
You wanted to merge `origin/master`, right? I think that went wrong.

**Comment by euronion** [comment]
Yes, I realised that. Since the branch was based on a now outdated master I wanted to merge and accidentally reverted the commit. Let's try that again.

**Pull request merged**